components/SendString.coffee

```diff
@@ -1,42 +1,22 @@
 noflo = require 'noflo'
 
-class SendString extends noflo.Component
-  constructor: ->
-    @data =
-      string: null
-      group: []
-    @groups = []
-    @inPorts = new noflo.InPorts
-      string:
-        datatype: 'string'
-      in:
-        datatype: 'bang'
-    @outPorts = new noflo.OutPorts
-      out:
-        datatype: 'string'
-
-    @inPorts.string.on 'data', (data) =>
-      @data.string = data
-
-    @inPorts.in.on 'begingroup', (group) =>
-      @groups.push group
-
-    @inPorts.in.on 'data', (data) =>
-      return if @data.string is null
-      @data.group = @groups.slice 0
-      @sendString @data
-
-    @inPorts.in.on 'endgroup', (group) =>
-      @groups.pop()
-    @inPorts.in.on 'disconnect', =>
-      @outPorts.out.disconnect()
-
-  sendString: (data) ->
-    for group in data.group
-      @outPorts.out.beginGroup group
-    @outPorts.out.send data.string
-    for group in data.group
-      @outPorts.out.endGroup()
-
-
-exports.getComponent = -> new SendString
+exports.getComponent = ->
+  c = new noflo.Component
+  c.description = 'Send a string when receiving a packet'
+
+  c.inPorts.add 'string',
+    datatype: 'string'
+    description: 'String to send'
+    control: true
+  c.inPorts.add 'in',
+    datatype: 'bang'
+    description: 'Send the string out'
+  c.outPorts.add 'out',
+    datatype: 'string'
+
+  c.process (input, output) ->
+    return unless input.has 'string', 'in'
+    data = input.get 'in'
+    return unless data.type is 'data'
```

Comment:
If this component is supposed to ignore brackets on `in` rather than forward them, it may be reasonable to disable bracket forwarding in its settings.